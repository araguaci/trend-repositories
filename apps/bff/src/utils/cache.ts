const LRU = require('lru-cache');

export let cache = null;

export const initCache = () => {
  const options = {
    max: 500,
    ttl: 86400 * 1000, // 1 day
  };

  cache = new LRU(options);
};
