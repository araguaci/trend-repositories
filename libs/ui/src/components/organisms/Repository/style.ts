import styled from 'styled-components';

export const StyledRepository = styled.div`
  padding: 16px;
  box-shadow: inset 0 -1px 0 #21262d;
`;

export const RepositoryTitleContainer = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 3px;
`;
