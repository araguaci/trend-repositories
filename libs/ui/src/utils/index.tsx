/* istanbul ignore file */
import { ThemeProvider } from 'styled-components';
import theme from '../theme';
import { ReactNode } from 'react';
import { render, act } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import { MockedResponse } from '@apollo/client/testing/core';
import { InMemoryCache } from '@apollo/client/cache';

export const renderWithProvider = (node: ReactNode) => {
  return render(<ThemeProvider theme={theme}>{node}</ThemeProvider>);
};

export const renderWithMockProvider = (
  node: ReactNode,
  cache: InMemoryCache,
  mocks: ReadonlyArray<MockedResponse>
) => {
  return render(
    <ThemeProvider theme={theme}>
      <MockedProvider cache={cache} mocks={mocks}>
        {node}
      </MockedProvider>
    </ThemeProvider>
  );
};

export const renderWithThemeProvider = (node: ReactNode) => {
  return render(<ThemeProvider theme={theme}>{node}</ThemeProvider>);
};

export const waitForResponse = async () =>
  await act(async () => await new Promise((resolve) => setTimeout(resolve, 0)));
export * from '@testing-library/react';
