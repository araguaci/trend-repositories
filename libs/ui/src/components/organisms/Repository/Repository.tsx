import { FC } from 'react';
import { GithubRepository } from '@trend-repositories/api-interfaces';
import { Button, Icon, Link, Text } from '@ui/atoms';
import { RepositoryInformation, Topics } from '@ui/molecules';
import { RepositoryTitleContainer, StyledRepository } from './style';

const Repository: FC<GithubRepository & { onStar: () => void }> = ({
  name,
  description,
  stars,
  forks,
  license,
  languageColor,
  language,
  url,
  topics,
  isStarred,
  onStar,
}) => {
  return (
    <StyledRepository data-cy="repository">
      <RepositoryTitleContainer>
        <Link href={url}>{name}</Link>
        <Button
          data-cy="star-button"
          data-testid="star-button"
          onClick={onStar}
          color={isStarred ? 'blue_2' : undefined}
        >
          <Icon name="star" />
          Star
        </Button>
      </RepositoryTitleContainer>

      <Text elementType="p" size="medium">
        {description}
      </Text>
      <Topics topics={topics} />
      <RepositoryInformation
        name={name}
        url={url}
        stars={stars}
        license={license}
        forks={forks}
        languageColor={languageColor}
        language={language}
      />
    </StyledRepository>
  );
};
export default Repository;
