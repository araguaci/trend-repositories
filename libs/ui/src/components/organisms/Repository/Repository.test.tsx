import { renderWithThemeProvider, screen } from '@ui/utils';
import Repository from './Repository';
import { fireEvent } from '@testing-library/react';

describe('Repository Component Test', () => {
  it('should be rendered correctly', () => {
    const handleStar = jest.fn();
    const { asFragment } = renderWithThemeProvider(
      <Repository
        id={1}
        name="test"
        description="test"
        topics={['topic1', 'topic2']}
        license="MIT"
        stars={100}
        forks={100}
        language="Javascript"
        languageColor="yellow"
        url="http://fake-url"
        onStar={handleStar}
      />
    );

    const starButton = screen.getByTestId('star-button');

    fireEvent.click(starButton);

    expect(handleStar).toHaveBeenCalled();
    expect(asFragment()).toMatchSnapshot();
  });
});
