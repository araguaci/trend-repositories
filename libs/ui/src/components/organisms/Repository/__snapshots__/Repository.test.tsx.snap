// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repository Component Test should be rendered correctly 1`] = `
<DocumentFragment>
  <div
    class="style__StyledRepository-gb28vz-0 dDctPs"
    data-cy="repository"
  >
    <div
      class="style__RepositoryTitleContainer-gb28vz-1 gmaiQM"
    >
      <a
        class="style__StyledLink-ikww7b-0 fUma-dV"
        href="http://fake-url"
      >
        test
      </a>
      <a
        class="style__StyledLink-ikww7b-0 fUma-dV style__StyledButton-gn6n84-0 eAVKfN"
        data-cy="star-button"
        data-testid="star-button"
      >
        <svg
          aria-hidden="true"
          class="star_svg__octicon star_svg__octicon-star"
          data-view-component="true"
          height="16"
          width="16"
        >
          <path
            d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.75.75 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694v.001z"
            fill-rule="evenodd"
          />
        </svg>
        Star
      </a>
    </div>
    <p
      class="style__StyledText-sc-1aawx67-0 keFQXl"
    >
      test
    </p>
    <div
      class="style__StyledTopics-sc-19kmfo9-0 jwmWIf"
    >
      <a
        class="style__StyledLink-ikww7b-0 fUma-dV style__StyledBadge-o2pv9p-0 klpobS"
        href="https://github.com/search?q=topic:topic1&type=Repositories"
      >
        topic1
      </a>
      <a
        class="style__StyledLink-ikww7b-0 fUma-dV style__StyledBadge-o2pv9p-0 klpobS"
        href="https://github.com/search?q=topic:topic2&type=Repositories"
      >
        topic2
      </a>
    </div>
    <div
      class="style__StyledRepositoryInformatin-sc-1fukv0g-0 cTyAeW"
    >
      <div
        class="style__StyledLanguage-x0ke4t-0 fRquPl"
      >
        <div
          class="style__LanguageColor-x0ke4t-1 bXOzcH"
          color="yellow"
        />
        <span
          class="style__StyledText-sc-1aawx67-0 kEyWhB"
        >
          Javascript
        </span>
      </div>
      <div
        class="style__Info-sc-1fukv0g-1 gYIRUG"
      >
        <svg
          aria-hidden="true"
          class="licence_svg__octicon licence_svg__octicon-law licence_svg__mr-1"
          data-view-component="true"
          fill="#cdcdcd"
          height="16"
          width="16"
        >
          <path
            d="M8.75.75a.75.75 0 0 0-1.5 0V2h-.984c-.305 0-.604.08-.869.23l-1.288.737A.25.25 0 0 1 3.984 3H1.75a.75.75 0 0 0 0 1.5h.428L.066 9.192a.75.75 0 0 0 .154.838l.53-.53-.53.53v.001l.002.002.002.002.006.006.016.015.045.04a3.514 3.514 0 0 0 .686.45A4.492 4.492 0 0 0 3 11c.88 0 1.556-.22 2.023-.454a3.515 3.515 0 0 0 .686-.45l.045-.04.016-.015.006-.006.002-.002.001-.002L5.25 9.5l.53.53a.75.75 0 0 0 .154-.838L3.822 4.5h.162c.305 0 .604-.08.869-.23l1.289-.737a.25.25 0 0 1 .124-.033h.984V13h-2.5a.75.75 0 0 0 0 1.5h6.5a.75.75 0 0 0 0-1.5h-2.5V3.5h.984a.25.25 0 0 1 .124.033l1.29.736c.264.152.563.231.868.231h.162l-2.112 4.692a.75.75 0 0 0 .154.838l.53-.53-.53.53v.001l.002.002.002.002.006.006.016.015.045.04a3.517 3.517 0 0 0 .686.45A4.492 4.492 0 0 0 13 11c.88 0 1.556-.22 2.023-.454a3.512 3.512 0 0 0 .686-.45l.045-.04.01-.01.006-.005.006-.006.002-.002.001-.002-.529-.531.53.53a.75.75 0 0 0 .154-.838L13.823 4.5h.427a.75.75 0 0 0 0-1.5h-2.234a.25.25 0 0 1-.124-.033l-1.29-.736A1.75 1.75 0 0 0 9.735 2H8.75V.75zM1.695 9.227c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L3 6.327l-1.305 2.9zm10 0c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L13 6.327l-1.305 2.9z"
            fill-rule="evenodd"
          />
        </svg>
        <span
          class="style__StyledText-sc-1aawx67-0 kEyWhB"
        >
          MIT
        </span>
      </div>
      <div
        class="style__Info-sc-1fukv0g-1 gYIRUG"
      >
        <a
          class="style__StyledLink-ikww7b-0 fUma-dV"
          data-testid="stargazers-url"
          href="http://fake-url/stargazers"
        >
          <svg
            aria-hidden="true"
            class="star_svg__octicon star_svg__octicon-star"
            data-view-component="true"
            fill="#cdcdcd"
            height="16"
            width="16"
          >
            <path
              d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.75.75 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694v.001z"
              fill-rule="evenodd"
            />
          </svg>
          <span
            class="style__StyledText-sc-1aawx67-0 kEyWhB"
          >
            100
          </span>
        </a>
      </div>
      <div
        class="style__Info-sc-1fukv0g-1 gYIRUG"
      >
        <a
          class="style__StyledLink-ikww7b-0 fUma-dV"
          data-testid="forks-url"
          href="http://fake-url/network/members.test"
        >
          <svg
            aria-label="fork"
            class="fork_svg__octicon fork_svg__octicon-repo-forked"
            data-view-component="true"
            fill="#cdcdcd"
            height="16"
            width="16"
          >
            <path
              d="M5 3.25a.75.75 0 1 1-1.5 0 .75.75 0 0 1 1.5 0zm0 2.122a2.25 2.25 0 1 0-1.5 0v.878A2.25 2.25 0 0 0 5.75 8.5h1.5v2.128a2.251 2.251 0 1 0 1.5 0V8.5h1.5a2.25 2.25 0 0 0 2.25-2.25v-.878a2.25 2.25 0 1 0-1.5 0v.878a.75.75 0 0 1-.75.75h-4.5A.75.75 0 0 1 5 6.25v-.878zm3.75 7.378a.75.75 0 1 1-1.5 0 .75.75 0 0 1 1.5 0zm3-8.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5z"
              fill-rule="evenodd"
            />
          </svg>
          <span
            class="style__StyledText-sc-1aawx67-0 kEyWhB"
          >
            100
          </span>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
